packages/launchpad/cypress/e2e/integration/choose-a-browser.spec.ts

```diff
@@ -0,0 +1,225 @@
+      cy.get('[data-cy="alert-suffix-icon"]').click()
+      cy.get('[data-cy="alert-header"]').should('not.exist')
+    })
+
+    it('shows installed browsers with their relevant properties', () => {
+      cy.openProject('launchpad', ['--e2e'])
+
+      cy.visitLaunchpad()
+
+      stepThroughConfigPages()
+
+      cy.get('h1').should('contain', 'Choose a Browser')
+
+      cy.findByRole('radio', { name: 'Chrome v1.x' })
+
+      cy.findByRole('radio', { name: 'Firefox v2.x' })
+
+      cy.findByRole('radio', { name: 'Electron v3.x' })
+
+      cy.findByRole('radio', { name: 'Edge v4.x' })
+    })
+
+    it('performs mutation to launch selected browser when launch button is pressed', () => {
+      cy.openProject('launchpad', ['--e2e'])
+
+      cy.visitLaunchpad()
+
+      stepThroughConfigPages()
+
+      cy.get('h1').should('contain', 'Choose a Browser')
+
+      cy.contains('button', 'Launch Chrome').as('launchButton')
+
+      // Stub out response to prevent browser launch but not break internals
+      cy.intercept('mutation-OpenBrowser_LaunchProject', {
+        body: {
+          data: {
+            launchOpenProject: true,
+            setProjectPreferences: {
+              currentProject: {
+                id: 'test-id',
+                title: 'launchpad',
+                __typename: 'CurrentProject',
+              },
+              __typename: 'Query',
+            },
+          },
+        },
+      }).as('launchProject')
+
+      cy.get('@launchButton').click()
+
+      cy.wait('@launchProject').then(({ request }) => {
+        expect(request?.body.variables.browserPath).to.contain('/test/chrome/path')
+        expect(request?.body.variables.testingType).to.eq('e2e')
+      })
+    })
+
+    it('performs mutation to change selected browser when browser item is clicked', () => {
+      cy.openProject('launchpad', ['--e2e'])
+
+      cy.visitLaunchpad()
+      stepThroughConfigPages()
+
+      cy.get('h1').should('contain', 'Choose a Browser')
+
+      cy.findByRole('radio', { name: 'Chrome v1.x', checked: true }).as('chromeItem')
+      cy.findByRole('radio', { name: 'Firefox v2.x', checked: false }).as('firefoxItem')
+
+      cy.contains('button', 'Launch Chrome').should('be.visible')
+
+      cy.intercept('mutation-OpenBrowserList_SetBrowser').as('setBrowser')
+
+      cy.get('@firefoxItem').then(($input) => {
+        cy.get(`label[for=${$input.attr('id')}]`).click().then(() => {
+          cy.wait('@setBrowser').its('request.body.variables.id').should('eq', $input.attr('id'))
+        })
+      })
+
+      cy.findByRole('radio', { name: 'Chrome v1.x', checked: false })
+      cy.findByRole('radio', { name: 'Firefox v2.x', checked: true })
+
+      cy.contains('button', 'Launch Firefox').should('be.visible')
+    })
+  })
+
+  describe('No System Browsers Detected', () => {
+    beforeEach(() => {
+      setupZeroMockBrowsers()
+    })
+
+    it('does not show browser list when no browsers are detected', () => {
```

Comment:
@tbiethman This is not the correct behavior. If no browsers are detected - the Electron browser should always be shown. Can you track down if this is just a mocking issue or the actual behavior? There's a line item in the spreadsheet that was added for this use case (it wasn't there when you started this work).

Reply:
This is a mocking issue. The server implementation that we are mocking here is responsible for ensuring that the electron browser is always present. That mocked interface is going to make this specific requirement hard to test e2e, but we can certainly update this scenario (or add another) to show only electron and have some coverage of this expected workflow.